NuguCore/Sources/Business/Network/Api/NuguApiProvider.swift

```diff
@@ -101,38 +102,40 @@ class NuguApiProvider: NSObject {
         var error: Error?
         
         return Single<Observable<Data>>.create { [weak self] (single) -> Disposable in
-            let disposable = Disposables.create()
-            guard let self = self else { return disposable }
-            
             // reset error
             error = nil
+            let disposable = Disposables.create()
+            guard let self = self else { return disposable }
             
-            // enable client side load balance and find new resource server for directive and event both.
-            self.isCSLBEnabled = true
-            
-            guard let resourceServerUrl = self.url else {
-                single(.error(NetworkError.noSuitableResourceServer))
-                return disposable
-            }
-            
-            if self.serverSideEventProcessor == nil {
-                self.serverSideEventProcessor = ServerSideEventProcessor()
+            self.processorQueue.async {
+                // enable client side load balance and find new resource server for directive and event both.
+                self.isCSLBEnabled = true
                 
-                // connect downstream.
-                guard let downstreamUrl = URL(string: resourceServerUrl+"/"+NuguApi.directives.path) else {
-                    log.error("invailid url: \(resourceServerUrl+"/"+NuguApi.directives.path)")
+                guard let resourceServerUrl = self.url else {
                     single(.error(NetworkError.noSuitableResourceServer))
-                    return disposable
+                    return
+                }
+                
+                if self.serverSideEventProcessor == nil {
```

Comment:
```suggestion
                guard let serverSideEventProcessor = self.serverSideEventProcessor else {
                    let processor = ServerSideEventProcessor()
                    self.serverSideEventProcessor = processor
                     // ...
                    single(.success(processor.subject))
                    return
                }
                
                single(.success(serverSideEventProcessor.subject))
```
가급적이면 implicitly 제거하는게 좋지 않을까요

Reply:
> 가급적이면 implicitly 제거하는게 좋지 않을까요

묵시적으로 동작하도록 작성된 코드가 어떤건지 잘 파악이 안되는데요..
혹시 아래 코드 말씀하시는건가요?

```
guard let serverSideEventProcessor = self.serverSideEventProcessor else {}
```

Reply:
> 가급적이면 implicitly 제거하는게 좋지 않을까요

제생각에는 바로 위에서 할당했기 때문에 괜찮을 것 같고, single호출부가 통일되어 기존이 더 나은 것 같습니다.

single호출부를 통일하기위해 guard로 바꾸지 않고 기존코드에서 `let processor = ServerSideEventProcessor()`으로 선언하고 `self.serverSideEventProcessor`에 할당할 수도 있겠으나, 큰 의미가 있지는 않을 것 같네요...

말씀하신대로 `Forced Unwrapping`을 가급적 사용하지 말아야 하겠습니다만, 같은 flow 내에서까지 제한하면 필요 없는 분기나 메모리할당이 늘어나게 되는 것 같습니다.